File: Front/.Net Maui/BetFriends.Infrastructure/Repository/InMemoryFriendRepository.cs
using BetFriend.Domain.Friends;
using BetFriends.Domain.Abstractions;
using BetFriends.Domain.Features.RetrieveFriends;
using BetFriends.Infrastructure.Repository;

namespace BetFriend.Infrastructure.Repository;

internal class InMemoryFriendRepository : IFriendRepository
{
    private readonly List<string> friends = new();
    private readonly InMemoryMemberRepository memberRepository;

    public InMemoryFriendRepository(InMemoryMemberRepository memberRepository)
    {
        this.memberRepository = memberRepository;
    }
    public Task AddAsync(string id, CancellationToken cancellationToken)
    {
        friends.Add(id);
        var member = memberRepository.Members.First(x => x.MemberId == id);
        member.IsFriend = true;
        return Task.CompletedTask;
    }

    public async Task<IReadOnlyCollection<FriendDto>> GetFriendsAsync()
    {
        var friends = (await memberRepository.RetrieveMembersByKeyword(string.Empty)).Where(x => x.IsFriend);
        return friends.Select(x => new FriendDto(x.MemberId, x.Name)).ToList();
    }
}

File: Front/TS/src/infrastructure/adapters/repository/InMemoryFriendRepository.ts
import { FriendDto } from '../../../domain/features/RetrieveFriendsHandler';
import { IFriendRepository } from "../../../domain/friends/IFriendRepository";
import { IMemberRepository, MemberDto } from '../../../domain/members/IMemberRepository';

export class InMemoryFriendRepository implements IFriendRepository {

    private friends: string[] = [];
    constructor(private memberRepository: IMemberRepository) {}
    async GetFriendsAsync(): Promise<FriendDto[]> {
        const members = await this.memberRepository.RetrieveByKeywordAsync("");
        return members.filter(x => x.IsFriend).map<FriendDto>((x) => ({
            Id: x.Id,
            Name: x.Name
        }) )
    }

    SaveAsync(memberId: string): Promise<void> {
        this.friends.push(memberId);
        return Promise.resolve();
    }

}